src/tooltip/Tooltip.tsx

```diff
@@ -1,357 +1,223 @@
-          {popover}
-        </View>
-      </View>
-    );
-  };
-
-  componentDidMount() {
-    this._isMounted = true;
-    // wait to compute onLayout values.
-    requestAnimationFrame(this.getElementPosition);
-  }
-  componentWillUnmount() {
-    this._isMounted = false;
-  }
-
-  getElementPosition = () => {
-    const { skipAndroidStatusBar } = this.props;
-    this.renderedElement &&
-      this.renderedElement.measure(
-        (
-          _frameOffsetX,
-          _frameOffsetY,
-          width,
-          height,
-          pageOffsetX,
-          pageOffsetY
-        ) => {
-          this._isMounted &&
-            this.setState({
-              xOffset: pageOffsetX,
-              yOffset:
-                isIOS || skipAndroidStatusBar
-                  ? pageOffsetY
-                  : pageOffsetY -
-                    Platform.select({
-                      android: StatusBar.currentHeight,
-                      ios: 20,
-                      default: 0,
-                    }),
-              elementWidth: width,
-              elementHeight: height,
-            });
-        }
-      );
-  };
-
-  renderStaticModalContent = () => {
-    const { withOverlay, overlayColor } = this.props;
-    return (
-      <Fragment>
-        <TouchableOpacity
-          style={this.containerStyle(withOverlay, overlayColor)}
-          onPress={this.toggleTooltip}
-          activeOpacity={1}
-        />
-        <View>{this.renderContent(true)}</View>
-      </Fragment>
-    );
-  };
-
-  renderTogglingModalContent = () => {
-    const { withOverlay, overlayColor } = this.props;
-    return (
-      <TouchableOpacity
-        style={this.containerStyle(withOverlay, overlayColor)}
-        onPress={this.toggleTooltip}
-        activeOpacity={1}
-      >
-        {this.renderContent(true)}
-      </TouchableOpacity>
-    );
+  const calcDim = () => {
+    requestAnimationFrame(() => {
+      containerElement.current?.measure((_x, _y, width, height, px, py) => {
+        setContainerDimensions({ width, height, px, py });
+      });
+      tooltipElement.current?.measure((_x, _y, width, height) => {
+        setTooltipDimension({ width, height });
+      });
+    });
   };
-
-  renderModalContent = () => {
-    const { closeOnlyOnBackdropPress } = this.props;
-    if (closeOnlyOnBackdropPress) {
-      return this.renderStaticModalContent();
-    } else {
-      return this.renderTogglingModalContent();
+  React.useEffect(() => {
+    Animated.timing(animation, {
+      toValue: Number(visible),
+      useNativeDriver: true,
+      duration: 200,
+    }).start();
+  }, [visible, animation]);
+
+  React.useEffect(() => {
+    Dimensions.addEventListener('change', calcDim);
+    return () => {
+      Dimensions.removeEventListener('change', calcDim);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    calcDim();
+  }, [visible]);
+
+  const ToolTipStyle = () => {
+    const { px, width, height, py } = containerDimensions;
```

Comment:
I recommend using platfrom inputs for this decisioning 
https://github.com/react-native-elements/react-native-elements/blob/91d582a19ba3e34b078740fc374a20a8938a3f6b/src/tooltip/Tooltip.tsx#L284

Reply:
Done